synapse/handlers/sync.py

```diff
@@ -1440,18 +1440,22 @@ async def generate_sync_result(
 
         logger.debug("Fetching room data")
 
+        block_all_presence_data = (
+            since_token is None and sync_config.filter_collection.blocks_all_presence()
+        )
```

Comment:
> I'm slightly surprised that there's an `and` here rather than an `or`. I wonder if this is correct?

I did not change it in this pull request (as you allude to), it does seem fishy though. I don't understand why we don't return certain things on initial sync TBH.

> Additionally: IIRC there is a Synapse config option to disable presence. I wonder if that should be taken into account here?

Ah, the line below this takes that into account by checking the preference (see below). I think it would be more correct to group that into the single variable though -- otherwise we still risk doing some calculations which will be thrown away.

https://github.com/matrix-org/synapse/blob/f398886ab840061952a4d96ffe2261369c90640a/synapse/handlers/sync.py#L1462-L1471

Reply:
I did the refactor for the config flag and also inverted the logic so it is "include" instead of "block all" (which makes my head hurt less) as a separate PR: #14970.